server/src/main/java/org/elasticsearch/transport/InboundHandler.java

```diff
@@ -325,13 +338,17 @@ private static void sendErrorResponse(String actionName, TransportChannel transp
             response = handler.read(stream);
             response.remoteAddress(remoteAddress);
         } catch (Exception e) {
-            final Exception serializationException = new TransportSerializationException(
-                "Failed to deserialize response from handler [" + handler + "]",
-                e
-            );
-            logger.warn(new ParameterizedMessage("Failed to deserialize response from [{}]", remoteAddress), serializationException);
-            handleException(handler, serializationException);
-            return;
+            try {
+                final Exception serializationException = new TransportSerializationException(
+                    "Failed to deserialize response from handler [" + handler + "]",
+                    e
+                );
+                logger.warn(new ParameterizedMessage("Failed to deserialize response from [{}]", remoteAddress), serializationException);
+                handleException(handler, serializationException);
```

Comment:
can we not avoid the try-catch by just asserting before calling `handleException`?